### PR TITLE
Updates the README with descriptions of Orville some sample code

### PR DESCRIPTION
**Pull request by qxjit**
This is probably easiest to review by browsing files on the commit to merge here: https://github.com/flipstone/orville/tree/42fb5446cb5dd5f394cac1480b7fcf28642bb7e9

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,4 +1,40 @@
...
+extensions. It strikes a balance between enforcing type-safety in database
+interactions where it is reasonable and presenting type signatures that are
+minimally complicated.
+
+<br clear="left"/>
+
+## Why Orville?
+
+Orville is not meant to replace existing PostgreSQL libraries in the Haskell
+ecosystem, but to complement them. It has the power to satsify most experienced
```
Comment (AugmenTab) [inline docs/README.md:20]: satsify -> satisfy

File: docs/README.md [hunk 2]
```diff
@@ -1,4 +1,40 @@
...
+Haskell developers but strives to remain approachable to newcomers despite
+this. Orville's API is rich enough to be used in production on large and
+sophisticated applications, but avoids complicated type-level programming. If
+your application is too large to reasonably write all your SQL statements by
+hand yet doesn't require absolute type-safety between your custom SQL
+statements result sets and the Haskell types they decode into, Orville may be
```
Comment (AugmenTab) [inline docs/README.md:26]: Did you mean "your custom SQL statements' result sets" here?

File: docs/README.md [hunk 3]
```diff
@@ -9,3 +45,116 @@ See the tutorials, in order of increasing complexity:
 * [Using JSON](../JSON.md)
 
 Additional documentation is available in the Haddocks.
+
+## Just show me some code!
+
+Ok! Here's a very simple application that inserts to entities of a `Pet` model and finds one of them based on its name.
```
Comment (AugmenTab) [inline docs/README.md:51]: I think the "to" here is unnecessary.

File: docs/README.md [hunk 4]
```diff
@@ -9,3 +45,116 @@ See the tutorials, in order of increasing complexity:
...
+data Pet =
+  Pet
+    { petId :: PetId
+    , petName :: T.Text
+    }
+
+{- |
+  It's good practice to create newtype specific to each entity to hold its
+  primary key value
+-}
```
Comment (AugmenTab) [inline docs/README.md:74]: newtype -> newtypes; missing period at end of comment.

File: docs/README.md [hunk 5]
```diff
@@ -9,3 +45,116 @@ See the tutorials, in order of increasing complexity:
...
+  field to and from.
+-}
+nameField :: O.FieldDefinition O.NotNull T.Text
+nameField =
+  O.unboundedTextField "name"
+
+{- |
+  Marshaller above is associated with the @pet@ table. The marshallers fields
+  will define the column of the table.
+-}
```
Comment (AugmenTab) [inline docs/README.md:105]: marshallers -> marshaller's; column -> columns

File: docs/README.md (hunk 2 above)
Comment (qxjit) [reply docs/README.md:26]: Really I think it should be "your customs SQL statements, their result sets and...". Thanks!

File: docs/README.md (hunk 3 above)
Comment (qxjit) [reply docs/README.md:51]: You're right. It should probably say "inserts some entities" though rather than just "inserts entities"

**Review by AugmenTab — APPROVED** [review]

**Pull request merged**